### ISSUE TITLE
Audit columns, before insert/update events.

### ISSUE THREAD

**Issue by AndrePScope** [linq2db/linq2db#2142]
I need to implement the simple way of audit for a table, to support some system columns: Created, Updated.

Can I set these columns somewhere in **one place** on the client-side via linq2db, not on DB-side?
F.e. to have some events "OnBeforeUpdate"/"OnBeforeInset" where I can implement:
OnBeforeInset:  

> this.Created = DateTime.Now;

OnBeforeUpdate:  

> this.Updated = DateTime.Now;

I could not find such a possibility in linq2db... Is it so?

P.S. I cannot do this on DB-side only via default values ("updated" should be empty in insert).
And unfortunately, trigger creation does not meet company standards.

**Comment by Franki1986** [comment]
You could ovveride the ProcessQuery and than handle an update or insert. I did this to truncate strings that are too long: [Truncate strings](https://github.com/linq2db/linq2db/issues/2028#issuecomment-575545388)

**Comment by sdanyliv** [comment]
No universal way now. I suggest to write yourself new extensions:
```cs
UpdateAudited(obj)
InsertAudited(obj)
```
And for queries:
```cs
query.Set(...)
   .UpdateAudited() // this should add new Set automatically

```

**Comment by AndrePScope** [comment]
Thank you, guys!
@Franki1986, your suggestion is quite well !!! Thank you very much!

**Comment by AndrePScope** [comment]
Hi again!

Could you provide me with this:
>query.Set(...)
   .UpdateAudited() // this should add new Set automatically

Can I get access to the DbContext (Connection) in UpdateAudited?
(I need this because my DbContext keeps the information about currentUser that needed in audit)
I have only  IUpdatable<T> here... and looks have not the access to the Connection.

I know that another way is pass this info directly to the UpdateAudited...
But it looks not quite well...

**Comment by sdanyliv** [comment]
For now it is possible only via IQueryable:
```cs
LinqToDB.Linq.Internals.GetDataContext(query)
```

I will add additional extensions for `IUpdatable<>` and `IValueInsertable<>` in 3.0.preview-2
Or you can do it yourself via reflection.

**Comment by AndrePScope** [comment]
@sdanyliv, thank you!

So, I will wait for  3.0.preview-2. 
But as for now, I will implement this temporary via reflection.

Thank you!

**Comment by sdanyliv** [comment]
Fixed in #2148
